dojo/tools/sarif/parser.py

```diff
@@ -149,10 +138,92 @@ def cve_try(val):
         return None
 
 
+def get_title(result, rule):
+    title = get_message_from_multiformatMessageString(result['message'], rule)
+    if rule is not None:
+        if title is None and 'shortDescription' in rule:
+            title = get_message_from_multiformatMessageString(rule['shortDescription'], rule)
+        elif title is None and 'fullDescription' in rule:
+            title = get_message_from_multiformatMessageString(rule['fullDescription'], rule)
+        elif title is None and 'name' in rule:
+            title = rule['name']
+        elif title is None:
+            title = rule['id']
+
+    if title is None:
+        title = ''
+
+    return textwrap.shorten(title, 150)
+
+
+def get_snippet(result):
+    snippet = None
+    if 'locations' in result:
+        location = result['locations'][0]
+        if 'physicalLocation' in location:
+            if 'region' in location['physicalLocation']:
+                if 'snippet' in location['physicalLocation']['region']:
+                    if 'text' in location['physicalLocation']['region']['snippet']:
+                        snippet = location['physicalLocation']['region']['snippet']['text']
+    return snippet
+
+
+def get_description(result, rule):
+    description = ''
+    message = ''
+    if 'message' in result:
+        message = get_message_from_multiformatMessageString(result['message'], rule)
+        description += '**Result message:** {}\n'.format(message)
+    if get_snippet(result) is not None:
+        description += '**Snippet:** {}\n'.format(get_snippet(result))
```

Comment:
We need to add ``` to escape the snippet code.

Reply:
Very good idea, done.

Reply:
I think we need some `\n` chars to make it works as expected.

Reply:
Where do you need the `\n` characters? In one of my njsscan examples the snippet is

```
"snippet": {
    "text": "      await db.query(sql`\n        INSERT INTO users (id, email, first_name, last_name, password_hash)\n        VALUES (${join([\n          user.id,\n          user.email,\n          user.firstName,\n          user.lastName,\n          user.passwordHash,\n        ])})\n      `)"
    },
```

The description in the database is

```
**Result message:** Untrusted input concatinated with raw SQL query can result in SQL Injection.
**Snippet:** ```      await db.query(sql`
        INSERT INTO users (id, email, first_name, last_name, password_hash)
        VALUES (${join([
          user.id,
          user.email,
          user.firstName,
          user.lastName,
          user.passwordHash,
        ])})
      `)```
**Rule name:** NodeSqliInjection
```

In the UI it looks like this:
![2021-09-12 16_58_39-View Finding _ DefectDojo](https://user-images.githubusercontent.com/2698502/132992558-5bf0c569-58ea-4c39-b1c5-37ae850ac541.png)

It looks fine to me.

Reply:
Could you add some line feed to have the same look as for other parser (like the Dependency Check one ?)
![image](https://user-images.githubusercontent.com/1694940/132994744-1e720d3a-8088-462e-a85e-9801d0c2fad4.png)

Reply:
I see, makes sense. With an additional space it looks like this now:
![2021-09-12 20_35_57-View Finding _ DefectDojo](https://user-images.githubusercontent.com/2698502/132999342-ff515eff-9900-4463-95df-a4773fdce010.png)

Reply:
awesome!